lib/gltfPipeline.js

```diff
@@ -0,0 +1,108 @@
+var removeUnusedSkins = require('./removeUnusedSkins');
+var removeUnusedCameras = require('./removeUnusedCameras');
+var removeUnusedTextures = require('./removeUnusedTextures');
+var removeUnusedMeshes = require('./removeUnusedMeshes');
+var removeUnusedNodes = require('./removeUnusedNodes');
+var removeUnusedAccessors = require('./removeUnusedAccessors');
+var removeUnused = require('./removeUnused');
+var parseBinaryGltf = require('./parseBinaryGltf');
+var addPipelineExtras = require('./addPipelineExtras');
+var convertDagToTree = require('./convertDagToTree');
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+var removeUnusedVertices = require('./removeUnusedVertices');
+var OptimizationStatistics = require('./OptimizationStatistics');
+var writeGltf = require('./writeGltf');
+var writeBinaryGltf = require('./writeBinaryGltf');
+var readGltf = require('./readGltf');
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var defaultValue = Cesium.defaultValue;
+
+module.exports = {
+    processJSON : processJSON,
+    processFile : processFile,
+    processJSONToDisk : processJSONToDisk,
+    processFileToDisk : processFileToDisk
+};
+
+function processJSON(gltfWithExtras, options, callback) {
+    var stats = new OptimizationStatistics();
+
+    addDefaults(gltfWithExtras, stats);
+
+    removeUnused(gltfWithExtras, stats);
+    printStats(stats);
+
+    convertDagToTree(gltfWithExtras);
+
+    removeUnusedVertices(gltfWithExtras);
+
+    combineMeshes(gltfWithExtras);
+    combinePrimitives(gltfWithExtras);
+
+    //Run removeUnused stage again after all pipeline stages have been run to remove objects that become unused
+    removeUnused(gltfWithExtras);
+
+    callback(gltfWithExtras);
+}
+
+function processFile(inputPath, options, callback) {
+    readGltf(inputPath, function(gltf) {
+        processJSON(gltf, options, function(gltf) {
+            callback(gltf);
+        });
+    });
+}
+
+function processJSONToDisk(gltfWithExtras, outputPath, options, callback) {// isBinary, isEmbedded) {
+    processJSON(gltfWithExtras, function(gltfWithExtras) {
+        var isBinary = defaultValue(options.isBinary, false);
+        var isEmbedded = defaultValue(options.isEmbedded, false);
+        if (isBinary) {
+            writeBinaryGltf(gltfWithExtras, outputPath, true, callback);
+        } else {
+            writeGltf(gltfWithExtras, outputPath, isEmbedded, true, callback);
+        }
+    });
+}
+
+function processFileToDisk(inputPath, outputPath, options, callback) {//isBinary, isEmbedded) {
+    processFile(inputPath, options, function(gltf) {
+        var isBinary = defaultValue(options.isBinary, false);
+        var isEmbedded = defaultValue(options.isEmbedded, false);
+        if (isBinary) {
+            writeBinaryGltf(gltf, outputPath, true, callback);
+        } else {
+            writeGltf(gltf, outputPath, isEmbedded, true, callback);
+        }
```

Comment:
A lot of the code in `processFileToDisk` and `processJSONToDisk` can be factored out into a common function.